### PR TITLE
feat(abci): Add round to Prepare/ProcessProposal,  FinalizeBlock

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

For better transaction processing in same-block execution mode, ABCI clients need round number in PrepareProposal, ProcessProposal, and FinalizeBlock.

## What was done?

1. Added missing field
2. Updated tests
3. Updated kvstore app

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: internal/consensus/replay_test.go [hunk 1]
```diff
@@ -864,17 +865,22 @@ func buildAppStateFromChain(
 	switch mode {
 	case 0:
 		for i := 0; i < nBlocks; i++ {
-			state = applyBlock(ctx, t, blockExec, state, chain[i])
+			round := int32(0)
+			if i+1 < len(chain) {
+				round = chain[i+1].LastCommit.Round
```
Comment (shumkov) [inline internal/consensus/replay_test.go:870]: Is round important for replay blocks?

Comment (shumkov) [reply internal/consensus/replay_test.go:870]: Do you mean deterministic and consistent on the ABCI app level?

File: internal/consensus/state_test.go [hunk 2]
```diff
@@ -2197,6 +2201,79 @@ func TestExtendVote(t *testing.T) {
...
+	var round int32
+	for round = 0; round < 4; round++ {
+		t.Logf("Starting round %d", round)
+		// we are proposer, so expect proposal
+		if round%4 == 0 {
+			mockProposerApplicationCalls(t, m, round, false)
+		}
+
+		// start the test
+		if started == false {
```
Comment (shotonoff) [inline internal/consensus/state_test.go:2238]: you can simplify this code a bit, if replace the condition on `round == 0`, then `started` variable would not be needed

File: internal/state/execution_test.go [hunk 3]
```diff
@@ -275,7 +276,7 @@ func TestProcessProposal(t *testing.T) {
 		TxResults: txResults,
 		Status:    abci.ResponseProcessProposal_ACCEPT,
 	}, nil)
-	uncommittedState, err := blockExec.ProcessProposal(ctx, block1, state, true)
+	uncommittedState, err := blockExec.ProcessProposal(ctx, block1, 12, state, true)
```
Comment (shotonoff) [inline internal/state/execution_test.go:279]: is `12` magic number ? :)

File: proto/tendermint/abci/types.proto [hunk 4]
```diff
@@ -129,6 +129,8 @@ message RequestPrepareProposal {
   google.protobuf.Timestamp time = 6 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
   // Merkle root of the next validator set.
   bytes next_validators_hash = 7;
+  // Round number for the block.
+  int32 round = 8;
```
Comment (shotonoff) [inline proto/tendermint/abci/types.proto:133]: WDYT usage `1XX` position for `round`, the same as for other dash related fields?

**Review by shotonoff — APPROVED** [review]
🚀 LGTM
pease have a look at the comments

**Pull request merged**